alkos/kernel/include/kernel_assert.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef KERNEL_INCLUDE_ASSERT_HPP_
+#define KERNEL_INCLUDE_ASSERT_HPP_
+
+#include <defines.hpp>
+#include <panic.hpp>
+#include <todo.hpp>
+
+// ------------------------------
+// Int type asserts
+// ------------------------------
+
+#define FAIL_KERNEL(expr)                                                                                              \
+    KernelPanic("Assertion failed: " TOSTRING(expr) " at file: " __FILE__ " and line: " TOSTRING(__LINE__));
+
+/* c-like debug only assert */
+#ifdef NDEBUG
+
+#define ASSERT(expr)
+
+#else
+
+#define ASSERT(expr)                                                                                                   \
+    if (!(expr))                                                                                                       \
+    {                                                                                                                  \
+        FAIL_KERNEL(expr);                                                                                             \
+    }
+
+#endif // NDEBUG
+
+/* release assert */
+#define ASSERT_ALWAYS(expr)                                                                                            \
```

Comment:
I don't understand:
1. The purpose of this (The code for this is literally the same as normal ASSERT)
2. The semantics behind this. Assert always? Tf that even means. I don't know what you want this to be or wanted to achieve.

Reply:
Basic C assert works only in debug mode we also need something which works in release mode. Here I proposed ASSERT_ALWAYS which workws always no matter build type which is strongly correlated with usual assert. We can work on names for example ASSERT would work as this macro and ASSERT_ONLY_IN_DEBUG like C-style assert

Reply:
done